site/src/api/api.ts

```diff
@@ -1243,6 +1242,35 @@ const getMissingParameters = (
     missingParameters.push(parameter)
   }
 
+  // Check if parameter "options" changed and we can't use old build parameters.
```

Comment:
@BrunoQuaresma I may need your frontend wisdom. I battling the following edge case here:

0. We have a `foobar` string parameter with options: `aaa`, `bbb` (default), `ccc`, and the workspace built on it. The workspace uses `bbb` option.
1. Change the parameter option `bbb` to something else, let's say `ddd`.
2. Click "Update".

I can see the UpdateParametersDialog, but when I click the `Update` button, it gets back to it. If I pick something else, and then click back to the default, it works fine.

I looked deeper, and it seems that Formik doesn't set the internal `form.values`, even though `initialValues` are passed. I'm not sure how I can proceed here.

Reply:
Apparently, I got it fixes with `enableReinitialize`. Thanks for the suggestion!